packages/frontend/src/ee/features/aiCopilot/components/ChatElements/AiArtifactPanel.tsx

```diff
@@ -0,0 +1,111 @@
+import { parseVizConfig } from '@lightdash/common';
+import { Box, Center, Loader, Stack, Text } from '@mantine-8/core';
+import { IconExclamationCircle } from '@tabler/icons-react';
+import { memo, useMemo, type FC } from 'react';
+import MantineIcon from '../../../../../components/common/MantineIcon';
+import { useInfiniteQueryResults } from '../../../../../hooks/useQueryResults';
+import { useAiAgentArtifact } from '../../hooks/useAiAgentArtifacts';
+import { useAiAgentArtifactVizQuery } from '../../hooks/useOrganizationAiAgents';
+import { useAiAgentPageLayout } from '../../providers/AiLayoutProvider';
+import { AiChartVisualization } from './AiChartVisualization';
+import { ChatElementsUtils } from './utils';
+
+export const AiArtifactPanel: FC = memo(() => {
```

Comment:
I think we should pass the Artifact as prop here instead! this way we can avoid breaking the no-conditional hook rules
(we only render this component when there is one defined ayways)